src/scripts/main.js

```diff
@@ -1,3 +1,293 @@
+    arr.sort((a, b) => {
+      return formatSalary(a.cells[index].textContent)
+        - formatSalary(b.cells[index].textContent);
+    });
+  }
+
+  tableBody.append(...arr);
+}
+
+function sortDESC(index) {
+  if (typeof arr[0].cells[index].textContent === 'number') {
+    arr.sort((a, b) => b.cells[index].textContent - a.cells[index].textContent);
+  }
+
+  if (typeof arr[0].cells[index].textContent === 'string') {
+    arr.sort((a, b) => {
+      return b.cells[index].textContent
+        .localeCompare(a.cells[index].textContent);
+    });
+  }
+
+  if (arr[0].cells[index].textContent.includes('$')) {
+    arr.sort((a, b) => {
+      return formatSalary(b.cells[index].textContent)
+        - formatSalary(a.cells[index].textContent);
+    });
+  }
+
+  tableBody.append(...arr);
+}
+
+const form = document.createElement('form');
+
+form.className = 'new-employee-form';
+table.after(form);
+
+const input = document.createElement('input');
+let previousText;
+
+form.insertAdjacentHTML('beforeend', `
+  <label>
+    Name:
+    <input
+      name="name"
+      type="text"
+      required
+      data-qa="name"
+    >
+  </label>
+  <label>
+    Position:
+    <input
+      name="position"
+      type="text"
+      required
+      data-qa="position"
+    >
+  </label>
+  <label>
+    Office:
+    <select name="office" required data-qa="office">
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>
+    Age:
+    <input
+      name="age"
+      type="number"
+      required
+      data-qa="age"
+    >
+  </label>
+  <label>
+    Salary:
+    <input
+      name="salary"
+      type="number"
+      required
+      data-qa="salary"
+    >
+  </label>
+  <button type="submit">
+    Save to table
+  </button>
+`);
+
+form.addEventListener('submit', e => {
+  e.preventDefault();
+
+  const data = new FormData(form);
+  const tr = document.createElement('tr');
+  const formatter = new Intl.NumberFormat('en-US');
+  const salary = `$${formatter.format(data.get('salary'))}`;
+
+  tr.innerHTML = `
+    <td>${data.get('name')}</td>
+    <td>${data.get('position')}</td>
+    <td>${data.get('office')}</td>
+    <td>${data.get('age')}</td>
+    <td>${salary}</td>
+  `;
+
+  for (const letter of data.get('name')) {
+    if ('0123456789'.includes(letter)) {
+      pushNotification(150, 10, 'Error', 'Your data is incorrect', 'error');
+
+      return;
+    }
+  }
+
+  for (const letter of data.get('position')) {
+    if ('0123456789'.includes(letter)) {
+      pushNotification(150, 10, 'Error', 'Your data is incorrect', 'error');
+
+      return;
+    }
+  }
+
+  if (data.get('name').length < 4
+    || data.get('age') < 18
+    || data.get('age') > 90) {
+    pushNotification(150, 10, 'Error', 'Your data is incorrect', 'error');
+
+    return;
+  }
+
```

Comment:
https://github.com/mate-academy/js_employees_table_DOM/pull/251/files#r1037422419

If you need help ask in chat.